### PR TITLE
Re-roll of GPG commit signing

### PR DESCRIPTION
**Pull request by krobelus**
This is #73 with review comments applied and two behavior changes.

(the broken Mac tests were https://github.com/pytest-dev/pytest/issues/5802)

### CODE REVIEW

**Review by mystor — APPROVED** [review]
Thanks for the patches and sorry for the slow reviews!

File: gitrevise/odb.py [hunk 1]
```diff
@@ -274,10 +306,43 @@ def new_commit(
...
+        try:
+            gpg = run(
+                (self.gpg, "--status-fd=2", "-bsau", key_id),
+                stdout=PIPE,
+                stderr=PIPE,
+                input=buffer,
+                check=True,
+            )
+        except CalledProcessError as gpg:
+            print(gpg.stderr.decode(), file=sys.stderr, end="")
```
Comment (rwe) [inline gitrevise/odb.py:334]: This access to `gpg.stderr` will never work: if `run()` throws, `gpg` will not have been assigned a value other than the `None` above.

Comment (rwe) [reply gitrevise/odb.py:334]: Nevermind, I see now that `gpg` is aliased to the caught exception itself!

File: tests/conftest.py [hunk 2]
```diff
@@ -59,6 +61,12 @@ def repo(hermetic_seal):
         yield repo
 
 
+@pytest.fixture
+def short_tmpdir():
+    with tempfile.TemporaryDirectory() as tdir:
+        yield py.path.local(tdir)
```
Comment (rwe) [inline tests/conftest.py:67]: `py.path` is deprecated. Should this just use `from pathlib import Path` which is used elsewhere in this project?

File: tests/test_gpgsign.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        git -c commit.gpgSign=false commit --allow-empty -m 'commit 2'
+        git -c commit.gpgSign=false commit --allow-empty -m 'commit 3'
+        git -c commit.gpgSign=false commit --allow-empty -m 'commit 4'
+    """
+    )
+    main(["HEAD~~", "--gpg-sign"])
+    assert repo.get_commit("HEAD~~").gpgsig is not None
+    assert repo.get_commit("HEAD~").gpgsig is not None
+    assert repo.get_commit("HEAD").gpgsig is not None
```
Comment (rwe) [inline tests/test_gpgsign.py:65]: Something concerning about these tests: they only check round-tripping of `gitrevise`'s signature parsing/generation. There are no tests with `git verify-commit`, for example, to demonstrate that the signature is formatted as `git` itself expects.

File: gitrevise/odb.py (hunk 1 above)
Comment (krobelus) [reply gitrevise/odb.py:334]: heh, that's pretty clever :open_mouth:

File: tests/conftest.py (hunk 2 above)
Comment (krobelus) [reply tests/conftest.py:67]: Thanks, I didn't know! I remember I was just glad to get the macOS tests working somehow..

File: tests/test_gpgsign.py (hunk 3 above)
Comment (krobelus) [reply tests/test_gpgsign.py:65]: Yeah, we should definitely have at least one test that shows that we produce/consume the same signatures as Git.

**Pull request merged**